server/flow.go

```diff
@@ -0,0 +1,408 @@
+
+	return flow, nil
+}
+
+const (
+	stepWelcome                  flow.Name = "welcome"
+	stepServerVersionQuestion    flow.Name = "server-verstion-question"
+	stepConfluenceURL            flow.Name = "confluence-url"
+	stepOAuthInput               flow.Name = "oauth-input"
+	stepCSversionLessthan9       flow.Name = "server-version-less-than-9"
+	stepCSversionGreaterthan9    flow.Name = "server-version-greater-than-9"
+	stepAnnouncementQuestion     flow.Name = "announcement-question"
+	stepAnnouncementConfirmation flow.Name = "announcement-confirmation"
+	stepDone                     flow.Name = "done"
+	stepCancel                   flow.Name = "cancel"
+	stepOAuthConnect             flow.Name = "oauth-connect"
+
+	keyConfluenceURL     = "ConfluenceURL"
+	keyIsOAuthConfigured = "IsOAuthConfigured"
+	redirectURL          = "dummyRedirectURL" // will be added in oauth PR
+)
+
+func cancelButton() flow.Button {
+	return flow.Button{
+		Name:    "Cancel setup",
+		Color:   flow.ColorDanger,
+		OnClick: flow.Goto(stepCancel),
+	}
+}
+
+func (fm *FlowManager) stepCancel(command string) flow.Step {
+	return flow.NewStep(stepCancel).
+		Terminal().
+		WithText(fmt.Sprintf("Confluence integration setup has stopped. Restart setup later by running `/confluence %s`. Learn more about the plugin [here](https://mattermost.gitbook.io/plugin-confluence/).", command)).
+		WithColor(flow.ColorDanger)
+}
+
+func continueButtonF(f func(f *flow.Flow) (flow.Name, flow.State, error)) flow.Button {
+	return flow.Button{
+		Name:    "Continue",
+		Color:   flow.ColorPrimary,
+		OnClick: f,
+	}
+}
+
+func continueButton(next flow.Name) flow.Button {
+	return continueButtonF(flow.Goto(next))
+}
+
+func (fm *FlowManager) getBaseState() flow.State {
+	config := fm.getConfiguration()
+	isOAuthConfigured := config.ConfluenceOAuthClientID != "" || config.ConfluenceOAuthClientSecret != ""
+	return flow.State{
+		keyConfluenceURL:     config.ConfluenceURL,
+		keyIsOAuthConfigured: isOAuthConfigured,
+	}
+}
+
+func (fm *FlowManager) StartSetupWizard(userID string, delegatedFrom string) error {
+	state := fm.getBaseState()
+
+	err := fm.setupFlow.ForUser(userID).Start(state)
+	if err != nil {
+		return err
+	}
+
+	fm.client.Log.Debug("Started setup wizard", "userID", userID, "delegatedFrom", delegatedFrom)
+
+	fm.trackStartSetupWizard(userID, delegatedFrom != "")
+
+	return nil
+}
+
+func (fm *FlowManager) trackStartSetupWizard(userID string, fromInvite bool) {
+	fm.tracker.TrackUserEvent("setup_wizard_start", userID, map[string]interface{}{
+		"from_invite": fromInvite,
+		"time":        model.GetMillis(),
+	})
+}
+
+func (fm *FlowManager) trackCompleteSetupWizard(userID string) {
+	fm.tracker.TrackUserEvent("setup_wizard_complete", userID, map[string]interface{}{
+		"time": model.GetMillis(),
+	})
+}
+
+func (fm *FlowManager) stepWelcome() flow.Step {
+	welcomeText := ":wave: Welcome to your Confluence integration! [Learn more](https://mattermost.gitbook.io/plugin-confluence/)"
```

Comment:
That URL only has a link to the README, should we send people directly to the README instead? cc: @cwarnermm 

```suggestion
	welcomeText := ":wave: Welcome to your Confluence integration! [Learn more](https://github.com/mattermost-community/mattermost-plugin-confluence#readme)"
```